nixos/modules/services/networking/dnsmasq.nix

```diff
@@ -101,10 +101,12 @@ in
           BusName = "uk.org.thekelleys.dnsmasq";
           ExecStart = "${dnsmasq}/bin/dnsmasq -k --enable-dbus --user=dnsmasq -C ${dnsmasqConf}";
           ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
+          PrivateTmp = true;
+          ProtectSystem = true;
+          ProtectHome = true;
+          Restart = "always";
```

Comment:
How about `Restart = "on-failure";`?

Reply:
In general I think ```Restart = "on-failure";``` should be applied but as ```dnsmasq``` is a pretty critical piece of infrastructure, it should probably be running the whole time even if manually shut down.

You can very easily argue though that this is a policy we shouldn't be setting for our users.

Would you feel more comfortable if I made that a configuration setting?

Reply:
I don't think this needs to be configurable:

```
systemd.services.dnsmasq.serviceConfig.Restart = "always";
```

is simple enough.
Under which circumstances dnsmasq shuts down itself gracefully, when it should not? Sounds like the bug is somewhere else?

Reply:
Indeed - if it exits with 0, we probably still want it to restart - it should simply be running all the time. Bug or no bug.